services/jenkins/jenkins-build.service.js

```diff
@@ -1,107 +1,95 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const serverSecrets = require('../../lib/server-secrets')
+const Joi = require('joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const JenkinsBase = require('./jenkins-base')
+const {
+  buildTreeParamQueryString,
+  buildUrl,
+  queryParamSchema,
+} = require('./jenkins-common')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class JenkinsBuild extends LegacyService {
+// https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/BallColor.java#L56
+const colorStatusMap = {
+  red: 'failing',
+  red_anime: 'building',
+  yellow: 'unstable',
+  yellow_anime: 'building',
+  blue: 'passing',
+  blue_anime: 'building',
+  green: 'passing',
+  green_anime: 'building',
+  grey: 'not built',
+  grey_anime: 'building',
+  disabled: 'not built',
+  disabled_anime: 'building',
+  aborted: 'not built',
+  aborted_anime: 'building',
+  notbuilt: 'not built',
+  notbuilt_anime: 'building',
+}
+
+const schema = Joi.object({
+  color: Joi.allow(...Object.keys(colorStatusMap)).required(),
+}).required()
+
+module.exports = class JenkinsBuild extends JenkinsBase {
   static get category() {
     return 'build'
   }
 
+  static get defaultBadgeData() {
+    return {
+      label: 'build',
+    }
+  }
+
   static get route() {
     return {
       base: 'jenkins/s',
```

Comment:
I don't have any PR #'s handy, but we seem to be trending away from single letter aliases to more explicit route param names. 

I'd like to do the same thing here (`build` or `status` instead of `s`). Any thoughts/objections?

Reply:
Happy if you want to change it, but we'll want to maintain an alias/redirect for `/s` for existing users

We talked about formalising some of this stuff in docs in https://github.com/badges/shields/pull/3144#issuecomment-470665539 and then.. haven't got round to doing it yet. There's several things in that discussion that I want to get done at some point but haven't found the time. Having this stuff documented would make it easier when we're thinking

> wait, what **is** our convention for a "build status" URL?

:slightly_smiling_face:

Reply:
I'll leave it in place for now. I plan on using some of the common bits in here to refactor the Jenkins test service so I'll consider the alias redirecting in that one